src/__tests__/proposals.int.test.ts

```diff
@@ -61,23 +61,90 @@ describe('/proposals', () => {
           200,
           [
             {
-              id: 1,
-              externalId: 'proposal-1',
+              id: 2,
+              externalId: 'proposal-2',
               applicantId: 1,
               opportunityId: 1,
-              createdAt: '2525-01-01T00:00:05.000Z',
+              createdAt: '2525-01-01T00:00:06.000Z',
             },
             {
-              id: 2,
-              externalId: 'proposal-2',
+              id: 1,
+              externalId: 'proposal-1',
               applicantId: 1,
               opportunityId: 1,
-              createdAt: '2525-01-01T00:00:06.000Z',
+              createdAt: '2525-01-01T00:00:05.000Z',
             },
           ],
         );
     });
 
+    it('returns according to pagination parameters', async () => {
+      await db.sql('opportunities.insertOne', {
+        title: '🔥',
+      });
+      await db.sql('applicants.insertOne', {
+        externalId: '12345',
+        optedIn: 'true',
+      });
+      await Array.from(Array(20)).reduce(async (p, _, i) => {
+        await p;
+        await db.sql('proposals.insertOne', {
+          applicantId: 1,
+          externalId: `proposal-${i + 1}`,
+          opportunityId: 1,
+        });
+      }, Promise.resolve());
```

Comment:
I haven't seen this pattern before! It's a little hard to parse, although I think I got there.

If you don't mind a little code golf, how does this look to you?

```suggestion
      await Promise.all(Array.from(Array(20), (_, i) => db.sql('proposals.insertOne', {
          applicantId: 1,
          externalId: `proposal-${i + 1}`,
          opportunityId: 1,
        })));
```

A few lines of code shorter, avoids `reduce` which is a difficult function to reason about, and runs the `INSERT`s all in parallel instead of sequentially to hopefully save a bit of test time.

That said, what you have is fine, and I'm 100% okay with you silently resolving this discussion without any changes!

Reply:
The reason I did the obnoxious sequential version is because I (think?) I need the test to have deterministic IDs -- otherwise I won't be able to know for sure which rows are supposed to be returned.

That said, maybe the IDs would be deterministic in parallel!  I was just assuming it would introduce a race condition into the test.

Reply:
Except a matcher then wouldn't test the core assertion: that we're sorting by ID. Ugh. Okay, thanks for bearing with me, I'm all caught up now!

Reply:
FWIW what I really wanted to do was a for loop (I mean, want is a strong word), but our linter hate hate hated that.

I could just guiltlessly disable the linter since it's a test, and the thing becomes instantly legible.  Thoughts?

Reply:
Maybe step back and reconsider what we actually are trying to test, and if there are alternative approaches?

I think the intention of this test is to make sure of is that we're returning the correct subset of results. Except, we're relying on PostgreSQL to do that for us... is this a worthwhile test? Under what circumstances would it meaningfully fail?

Maybe it'd make more sense to test that PostgreSQL will accept the queries we're generating, that we are able to get as many pages as we expect, and that we're not introducing any off-by-one errors in the query construction. None of those require in-order proposal creation, or even a clean database!

Reply:
Well, as a for instance... I do want to be sure that our pagination implementation will constructs the right query to return the X most recently created proposals (not simply X proposals).

We're validating our query construction and then using the output of psql's engine (which we trust) to test that query.

At any rate, I know this is resolved; we can always rip out the test later as desired.